cloud-aws/src/test/java/com/sequenceiq/cloudbreak/cloud/aws/connector/resource/AwsCloudWatchServiceTest.java

```diff
@@ -207,4 +207,49 @@ void testDeleteCloudWatchAlarmsForSystemFailuresDoesNotBatchDeletionWhenDescribe
         assertEquals(deleteAlarmNames,
                 captorDelete.getAllValues().stream().map(DeleteAlarmsRequest::getAlarmNames).collect(Collectors.toList()));
     }
+
+    @Test
+    void testDeleteCloudWatchAlarmsWhenInstanceDeletedOnProvider() {
+        String alarm1Name = "i-1-Status-Check-Failed-System";
+        String alarm2Name = "i-2-Status-Check-Failed-System";
+        String deletedAlarmName = "i-deleted-Status-Check-Failed-System";
+        String instanceId1 = "i-1";
+        String instanceId2 = "i-2";
+        String deletedInstanceId = "i-deleted";
+        MetricAlarm alarm1 = mock(MetricAlarm.class);
+        MetricAlarm alarm2 = mock(MetricAlarm.class);
+        MetricAlarm deletedInstanceAlarm = mock(MetricAlarm.class);
+        DescribeAlarmsResult describeAlarmsResult = mock(DescribeAlarmsResult.class);
+        AmazonCloudWatchClient cloudWatchClient = mock(AmazonCloudWatchClient.class);
+        AwsCredentialView credentialView = mock(AwsCredentialView.class);
+        CloudStack stack = mock(CloudStack.class);
+        Group group = mock(Group.class);
+        Group deletedGroup = mock(Group.class);
+        CloudInstance instance1 = mock(CloudInstance.class);
+        CloudInstance instance2 = mock(CloudInstance.class);
+        CloudInstance deletedInstance = mock(CloudInstance.class);
+        List<Group> groups = List.of(group, deletedGroup);
```

Comment:
**suggestion:** Consider breaking up this initialization block by separating the declarations + initializations above from the remaining `when(...)` setup.